### ISSUE TITLE
Use a Cluster Singleton for the AuctionScheduler

### ISSUE THREAD

**Issue by TimMoore** [lagom/online-auction-java#27]
`AuctionScheduler` is not cluster aware: it will run its scheduled task on all nodes.

We should demonstrate use of a [cluster singleton](http://doc.akka.io/docs/akka/2.4/java/cluster-singleton.html) to ensure it only runs on one node.

(We should port this change to Scala as well.)

**Comment by jroper** [comment]
Technically there's no problem with this, the worst that will happen is that duplicate commands get sent to end the auction, but the auction entity is tolerant to this.

**Comment by TimMoore** [comment]
I agree that it isn't important in this specific case, but since this is the go-to example for scheduled tasks, and since many use cases _will_ require a cluster singleton, I still think it's worth doing here as an example for others.

**Comment by jroper** [comment]
For that we should probably provide an abstraction in Lagom, perhaps offering the ability to schedule something once on every node, to schedule something as a cluster singleton, and to distribute a list of scheduled tasks across all the nodes.
